File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.java
import java.util.*;

class Solution {
    public String solution(String[] survey, int[] choices) {
        String answer = "";
        char[][] types = {{'R', 'T'}, {'C', 'F'}, {'J', 'M'}, {'A', 'N'}};
        int[] scores = {0, 3, 2, 1, 0, 1, 2, 3};
        Map<Character, Integer> hm = new HashMap<>();
        
        for(char[] type : types){
            hm.put(type[0], 0);
            hm.put(type[1], 0);
        }
        
        for(int i = 0; i < survey.length; i++){
            char s = (choices[i] < 4) ? survey[i].charAt(0) : survey[i].charAt(1);
            hm.put(s, hm.get(s) + scores[choices[i]]);
        }
        
        for(char[] type : types)
            answer += (hm.get(type[0]) >= hm.get(type[1])) ? type[0] : type[1];
    
        return answer;
    }
}

File: 프로그래머스/1/118666. 성격 유형 검사하기/성격 유형 검사하기.py
from collections import defaultdict
def solution(survey, choices):
    answer = ""
    indicator = defaultdict(int)
    
    score = [0, 3, 2, 1, 0, 1, 2, 3]
    
    for i in range(len(survey)):
        if choices[i] <= 4:
            indicator[survey[i][0]] += score[choices[i]]
        else:
            indicator[survey[i][1]] += score[choices[i]]
    
    print(indicator)
    answer += "R" if indicator["R"] >= indicator["T"] else "T"
    answer += "C" if indicator["C"] >= indicator["F"] else "F"
    answer += "J" if indicator["J"] >= indicator["M"] else "M"
    answer += "A" if indicator["A"] >= indicator["N"] else "N"
        
    return answer
